packages/flutter/lib/src/widgets/sliver_explicit_extent_list.dart

```diff
@@ -0,0 +1,137 @@
+    super.key,
+    required NullableIndexedWidgetBuilder itemBuilder,
+    required this.itemExtentBuilder,
+    ChildIndexGetter? findChildIndexCallback,
+    int? itemCount,
+    bool addAutomaticKeepAlives = true,
+    bool addRepaintBoundaries = true,
+    bool addSemanticIndexes = true,
+  }) : super(delegate: SliverChildBuilderDelegate(
+    itemBuilder,
+    findChildIndexCallback: findChildIndexCallback,
+    childCount: itemCount,
+    addAutomaticKeepAlives: addAutomaticKeepAlives,
+    addRepaintBoundaries: addRepaintBoundaries,
+    addSemanticIndexes: addSemanticIndexes,
+  ));
+
+  /// A sliver that places multiple box children in a linear array along the main
+  /// axis.
+  ///
+  /// [SliverFixedExtentList] places its children in a linear array along the main
+  /// axis starting at offset zero and without gaps. Each child is forced to have
+  /// the returned extent of [itemExtentBuilder] in the main axis and the
+  /// [SliverConstraints.crossAxisExtent] in the cross axis.
+  ///
+  /// This constructor uses a list of [Widget]s to build the sliver.
+  SliverExplicitExtentList.list({
+    super.key,
+    required List<Widget> children,
+    required this.itemExtentBuilder,
+    bool addAutomaticKeepAlives = true,
+    bool addRepaintBoundaries = true,
+    bool addSemanticIndexes = true,
+  }) : super(delegate: SliverChildListDelegate(
+    children,
+    addAutomaticKeepAlives: addAutomaticKeepAlives,
+    addRepaintBoundaries: addRepaintBoundaries,
+    addSemanticIndexes: addSemanticIndexes,
+  ));
+
+  /// The children extent builder.
+  final ItemExtentGetter itemExtentBuilder;
+
+  @override
+  RenderSliverExplicitExtentList createRenderObject(BuildContext context) {
+    final SliverMultiBoxAdaptorElement element = context as SliverMultiBoxAdaptorElement;
+    return RenderSliverExplicitExtentList(childManager: element, itemExtentBuilder: itemExtentBuilder);
+  }
+
+  @override
+  void updateRenderObject(BuildContext context, RenderSliverExplicitExtentList renderObject) {
+    renderObject.itemExtentBuilder = itemExtentBuilder;
+  }
+}
+
+/// A sliver that places multiple box children with the explicit main axis extent in
+/// a linear array.
+class RenderSliverExplicitExtentList extends RenderSliverFixedExtentBoxAdaptor {
+  /// Creates a sliver that contains multiple box children that have a explicit
+  /// extent in the main axis.
+  ///
+  /// The [childManager] argument must not be null.
+  RenderSliverExplicitExtentList({
+    required super.childManager,
+    required ItemExtentGetter itemExtentBuilder,
+  }) : _itemExtentBuilder = itemExtentBuilder;
+
+  @override
+  ItemExtentGetter get itemExtentBuilder => _itemExtentBuilder;
+  ItemExtentGetter _itemExtentBuilder;
+  set itemExtentBuilder(ItemExtentGetter value) {
+    if (_itemExtentBuilder == value) {
+      return;
+    }
+    _itemExtentBuilder = value;
+    markNeedsLayout();
+  }
+
+  @override
+  double get itemExtent => double.nan;
```

Comment:
Could this be nullable? Or maybe we can add an assertion as the body of this getter that says this should not be used?

Reply:
Yes, I have change it to nullable and add a assertion at the beginning of `RenderSliverFixedExtentBoxAdaptor.performLayout()`
```dart
    assert((itemExtent != null && itemExtentBuilder == null) ||
        (itemExtent == null && itemExtentBuilder != null));
    assert(itemExtentBuilder != null || (itemExtent!.isFinite && itemExtent! >= 0));
```